### PR TITLE
V2 Card

### PR DESCRIPTION
**Pull request by praveen970**
V2Card implementation
BasicCard
FileCard
AnnouncementCard

### Screenshots
FileCard
AnnouncementCard

### CODE REVIEW

File: FluentUI.Demo/src/main/res/drawable/image_280.png
Comment (NamanPandey) [inline FluentUI.Demo/src/main/res/drawable/image_280.png:1]: Where is this used?

File: FluentUI.Demo/src/main/res/drawable/preview_cover.png
Comment (NamanPandey) [inline FluentUI.Demo/src/main/res/drawable/preview_cover.png:1]: Unused?

File: fluentui_controls/src/main/java/com/microsoft/fluentui/tokenized/controls/BasicCard.kt [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    val token = basicCardTokens
+        ?: FluentTheme.controlTokens.tokens[ControlTokens.ControlType.BasicCard] as BasicCardTokens
+    val basicCardInfo = BasicCardInfo(CardType.Elevated)
+    val cornerRadius = token.cornerRadius(basicCardInfo = basicCardInfo)
+    val backgroundColor = token.backgroundColor(basicCardInfo = basicCardInfo)
+    val elevation = token.elevation(basicCardInfo = basicCardInfo)
+    val borderColor = token.borderColor(basicCardInfo = basicCardInfo)
+    val borderStrokeWidth = token.borderStrokeWidth(basicCardInfo = basicCardInfo)
+    val shape = RoundedCornerShape(cornerRadius)
+    val absoluteElevation = LocalAbsoluteElevation.current + elevation
```
Comment (NamanPandey) [inline fluentui_controls/src/main/java/com/microsoft/fluentui/tokenized/controls/BasicCard.kt:42]: Material variables and may be affected by Material theming if users use it. Can we have either fluent equivalent or control this somehow?

File: fluentui_controls/src/main/java/com/microsoft/fluentui/tokenized/controls/BasicCard.kt [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    val elevationOverlay = LocalElevationOverlay.current
```
Comment (NamanPandey) [inline fluentui_controls/src/main/java/com/microsoft/fluentui/tokenized/controls/BasicCard.kt:43]: Same as above

File: fluentui_controls/src/main/java/com/microsoft/fluentui/tokenized/controls/BasicCard.kt [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+                shape
+            )
+            .border(
+                borderStrokeWidth, borderColor, shape
+            )
+            .clip(shape)
+            .then(
+                if (onClick != null) Modifier.clickable(
+                    interactionSource = interactionSource,
+                    indication = rememberRipple(),
```
Comment (NamanPandey) [inline fluentui_controls/src/main/java/com/microsoft/fluentui/tokenized/controls/BasicCard.kt:61]: Use LocalIndication.current as remember Ripple is a material function.

File: fluentui_core/src/main/java/com/microsoft/fluentui/theme/token/ControlTokens.kt [hunk 4]
```diff
@@ -5,6 +5,7 @@
 
 package com.microsoft.fluentui.theme.token
 
+import android.service.controls.Control
```
Comment (NamanPandey) [inline fluentui_core/src/main/java/com/microsoft/fluentui/theme/token/ControlTokens.kt:8]: Required?

File: fluentui_core/src/main/java/com/microsoft/fluentui/theme/token/controlTokens/AnnouncementCardTokens.kt [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+        )
+    }
+
+    @Composable
+    open fun titleSize(announcementCardInfo: BasicCardControlInfo): TextStyle {
+        return FluentTheme.aliasTokens.typography[AliasTokens.TypographyTokens.Body1Strong]
+    }
+
+    @Composable
+    open fun textSize(announcementCardInfo: BasicCardControlInfo): TextStyle {
```
Comment (NamanPandey) [inline fluentui_core/src/main/java/com/microsoft/fluentui/theme/token/controlTokens/AnnouncementCardTokens.kt:42]: Can we rename this to typography with appropriate prefix. Would be coherent all across.

File: fluentui_core/src/main/java/com/microsoft/fluentui/theme/token/controlTokens/AnnouncementCardTokens.kt [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+
+    @Composable
+    open fun titleColor(announcementCardInfo: BasicCardControlInfo): Color {
+        return FluentTheme.aliasTokens.neutralForegroundColor[AliasTokens.NeutralForegroundColorTokens.Foreground1].value(
+            themeMode = FluentTheme.themeMode
+        )
+    }
+
+    @Composable
+    open fun titleSize(announcementCardInfo: BasicCardControlInfo): TextStyle {
```
Comment (NamanPandey) [inline fluentui_core/src/main/java/com/microsoft/fluentui/theme/token/controlTokens/AnnouncementCardTokens.kt:37]: Same as below.

File: fluentui_core/src/main/java/com/microsoft/fluentui/theme/token/controlTokens/FileCardTokens.kt [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+        )
+    }
+
+    @Composable
+    open fun iconSize(fileCardInfo: BasicCardControlInfo): Dp {
+        return GlobalTokens.iconSize(GlobalTokens.IconSizeTokens.IconSize240)
+    }
+
+    @Composable
+    open fun textSize(fileCardInfo: BasicCardControlInfo): TextStyle {
```
Comment (NamanPandey) [inline fluentui_core/src/main/java/com/microsoft/fluentui/theme/token/controlTokens/FileCardTokens.kt:59]: Rename to Typography with appropriate prefix.

File: fluentui_core/src/main/java/com/microsoft/fluentui/theme/token/controlTokens/FileCardTokens.kt [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+        return FluentTheme.aliasTokens.typography[AliasTokens.TypographyTokens.Body2]
+    }
+
+    @Composable
+    open fun subTextSize(fileCardInfo: BasicCardControlInfo): TextStyle {
```
Comment (NamanPandey) [inline fluentui_core/src/main/java/com/microsoft/fluentui/theme/token/controlTokens/FileCardTokens.kt:64]: Same as above

File: fluentui_controls/src/main/java/com/microsoft/fluentui/tokenized/controls/AnnouncementCard.kt [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+import androidx.compose.ui.unit.Dp
+import com.microsoft.fluentui.theme.FluentTheme
+import com.microsoft.fluentui.theme.token.ControlTokens
+import com.microsoft.fluentui.theme.token.StateColor
+import com.microsoft.fluentui.theme.token.controlTokens.*
+
+@Composable
+fun AnnouncementCard(
+    title: String,
+    text: String,
```
[the remark this answers is not in the document]
Comment (praveen970) [reply fluentui_controls/src/main/java/com/microsoft/fluentui/tokenized/controls/AnnouncementCard.kt:26]: changed to description. Description looks better

File: FluentUI.Demo/src/main/res/drawable/preview_cover.png
Comment (praveen970) [reply FluentUI.Demo/src/main/res/drawable/preview_cover.png:1]: Removed

File: fluentui_core/src/main/java/com/microsoft/fluentui/theme/token/ControlTokens.kt (hunk 4 above)
Comment (praveen970) [reply fluentui_core/src/main/java/com/microsoft/fluentui/theme/token/ControlTokens.kt:8]: Removed

File: fluentui_controls/src/main/java/com/microsoft/fluentui/tokenized/controls/FileCard.kt [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+fun FileCard(
+    text: String,
+    subText: String,
+    modifier: Modifier = Modifier,
+    interactionSource: MutableInteractionSource = remember { MutableInteractionSource() },
+    onClick: (() -> Unit)? = null,
+    previewImageVector: ImageVector? = null,
+    @DrawableRes previewImageDrawable: Int? = null,
+    textIcon: FluentIcon,
+    actionOverflowIcon: FluentIcon? = null,
```
Comment (mishramayank1) [inline fluentui_controls/src/main/java/com/microsoft/fluentui/tokenized/controls/FileCard.kt:40]: actionOverflowIconOnClick could be provided on FluentIcon

File: fluentui_controls/src/main/java/com/microsoft/fluentui/tokenized/controls/FileCard.kt [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+@Composable
+fun FileCard(
+    text: String,
+    subText: String,
+    modifier: Modifier = Modifier,
+    interactionSource: MutableInteractionSource = remember { MutableInteractionSource() },
+    onClick: (() -> Unit)? = null,
+    previewImageVector: ImageVector? = null,
+    @DrawableRes previewImageDrawable: Int? = null,
+    textIcon: FluentIcon,
```
Comment (mishramayank1) [inline fluentui_controls/src/main/java/com/microsoft/fluentui/tokenized/controls/FileCard.kt:39]: Should be ImageVector

File: fluentui_controls/src/main/java/com/microsoft/fluentui/tokenized/controls/FileCard.kt [hunk 12]
```diff
@@ -0,0 +1,136 @@
...
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.text.TextStyle
+import com.microsoft.fluentui.theme.FluentTheme
+import com.microsoft.fluentui.theme.token.ControlTokens
+import com.microsoft.fluentui.theme.token.FluentIcon
+import com.microsoft.fluentui.theme.token.Icon
+import com.microsoft.fluentui.theme.token.controlTokens.FileCardInfo
+import com.microsoft.fluentui.theme.token.controlTokens.FileCardTokens
+
+@Composable
```
Comment (mishramayank1) [inline fluentui_controls/src/main/java/com/microsoft/fluentui/tokenized/controls/FileCard.kt:30]: Add API documentation.

File: fluentui_controls/src/main/java/com/microsoft/fluentui/tokenized/controls/FileCard.kt [hunk 13]
```diff
@@ -0,0 +1,136 @@
...
+                    Image(
+                        painterResource(id = previewImageDrawable), contentDescription = ""
+                    )
+                } else if (previewImageVector != null) {
+                    Image(
+                        imageVector = previewImageVector,
+                        contentDescription = ""
+                    )
+                }
+                Row(
```
Comment (mishramayank1) [inline fluentui_controls/src/main/java/com/microsoft/fluentui/tokenized/controls/FileCard.kt:83]: Lets consider FileCard in 2 part : Preview & Info. And take padding of infoSection as one token.

File: fluentui_controls/src/main/java/com/microsoft/fluentui/tokenized/controls/FileCard.kt [hunk 14]
```diff
@@ -0,0 +1,136 @@
...
+                            style = subTextTypography.merge(TextStyle(color = subTextColor))
+                        )
+                    }
+                }
+            }
+            if (isPreviewAvailable && actionOverflowIcon != null) {
+                Box(
+                    modifier = Modifier
+                        .align(Alignment.TopEnd)
+                        .padding(end = actionOverflowPadding, top = actionOverflowPadding)
```
Comment (mishramayank1) [inline fluentui_controls/src/main/java/com/microsoft/fluentui/tokenized/controls/FileCard.kt:111]: actionOverFlowPadding should be of type Padding.

File: fluentui_controls/src/main/java/com/microsoft/fluentui/tokenized/controls/AnnouncementCard.kt [hunk 15]
```diff
@@ -0,0 +1,127 @@
...
+        modifier = modifier,
+        basicCardTokens = CustomBasicCardTokens() as BasicCardTokens
+    ) {
+        Box(
+            modifier = Modifier
+                .padding(all = previewPadding)
+                .semantics(mergeDescendants = true) {},
+            contentAlignment = Alignment.Center
+        ) {
+            Column(
```
Comment (mishramayank1) [inline fluentui_controls/src/main/java/com/microsoft/fluentui/tokenized/controls/AnnouncementCard.kt:78]: merge Box & Column as the Box has only one element

Comment (mishramayank1) [reply fluentui_controls/src/main/java/com/microsoft/fluentui/tokenized/controls/AnnouncementCard.kt:78]: Use one column and arrange image, text, button by applying their individual padding token & spacing tokens in between.

File: fluentui_controls/src/main/java/com/microsoft/fluentui/tokenized/controls/AnnouncementCard.kt [hunk 16]
```diff
@@ -0,0 +1,141 @@
...
+                .width(IntrinsicSize.Min)
+                .padding(previewPadding)
+                .semantics(mergeDescendants = true) {},
+            horizontalAlignment = Alignment.CenterHorizontally,
+            verticalArrangement = Arrangement.Center
+        ) {
+            Box(
+                modifier = Modifier
+                    .clip(RoundedCornerShape(previewCornerRadius))
+                    .padding(bottom = previewInfoPadding)
```
Comment (mishramayank1) [inline fluentui_controls/src/main/java/com/microsoft/fluentui/tokenized/controls/AnnouncementCard.kt:98]: remove padding & use Spacer

File: fluentui_controls/src/main/java/com/microsoft/fluentui/tokenized/controls/AnnouncementCard.kt [hunk 17]
```diff
@@ -0,0 +1,141 @@
...
+                        imageVector = previewImageVector,
+                        contentDescription = ""
+                    )
+                }
+            }
+            BasicText(
+                modifier = Modifier.padding(
+                    start = textHorizontalPadding,
+                    end = textHorizontalPadding,
+                    bottom = titleTextPadding
```
Comment (mishramayank1) [inline fluentui_controls/src/main/java/com/microsoft/fluentui/tokenized/controls/AnnouncementCard.kt:115]: user spacer for bottom

File: fluentui_controls/src/main/java/com/microsoft/fluentui/tokenized/controls/AnnouncementCard.kt [hunk 18]
```diff
@@ -0,0 +1,141 @@
...
+                ),
+                text = title,
+                style = titleTypography.merge(TextStyle(color = titleColor))
+                    .merge(TextStyle(textAlign = TextAlign.Center))
+            )
+            BasicText(
+                modifier = Modifier.padding(
+                    start = textHorizontalPadding,
+                    end = textHorizontalPadding,
+                    bottom = textButtonPadding
```
Comment (mishramayank1) [inline fluentui_controls/src/main/java/com/microsoft/fluentui/tokenized/controls/AnnouncementCard.kt:125]: user spacer for bottom

File: FluentUI.Demo/src/androidTest/java/com/microsoft/fluentuidemo/demos/V2CardUITest.kt [hunk 19]
```diff
@@ -0,0 +1,116 @@
...
+                    subText = "SubText",
+                    textIcon = Icons.Outlined.Call,
+                    actionOverflowIcon = FluentIcon(
+                        Icons.Outlined.MoreVert,
+                        contentDescription = "Options"
+                    )
+                )
+            }
+        }
+        composeTestRule.onRoot(useUnmergedTree = true).printToLog("Tree")
```
Comment (NamanPandey) [inline FluentUI.Demo/src/androidTest/java/com/microsoft/fluentuidemo/demos/V2CardUITest.kt:55]: Do we need this?

File: fluentui_core/src/main/java/com/microsoft/fluentui/theme/token/controlTokens/AnnouncementCardTokens.kt [hunk 20]
```diff
@@ -0,0 +1,116 @@
...
+
+data class AnnouncementCardInfo(
+    override val cardType: CardType = CardType.Elevated,
+    val buttonStyle: ButtonStyle = ButtonStyle.TextButton
+) : BasicCardControlInfo()
+
+@Parcelize
+open class AnnouncementCardTokens : BasicCardTokens(), Parcelable {
+    @Composable
+    open fun textColor(announcementCardInfo: BasicCardControlInfo): Color {
```
Comment (mishramayank1) [inline fluentui_core/src/main/java/com/microsoft/fluentui/theme/token/controlTokens/AnnouncementCardTokens.kt:24]: use the AnnouncementCardInfo

File: fluentui_core/src/main/java/com/microsoft/fluentui/theme/token/controlTokens/FileCardTokens.kt [hunk 21]
```diff
@@ -0,0 +1,104 @@
...
+
+data class FileCardInfo(
+    val isPreviewAvailable: Boolean = true,
+    override val cardType: CardType = CardType.Elevated
+) : BasicCardControlInfo()
+
+@Parcelize
+open class FileCardTokens : BasicCardTokens(), Parcelable {
+    @Composable
+    open fun iconColor(fileCardInfo: BasicCardControlInfo): Color {
```
Comment (mishramayank1) [inline fluentui_core/src/main/java/com/microsoft/fluentui/theme/token/controlTokens/FileCardTokens.kt:23]: use the FileCardInfo

File: fluentui_core/src/main/java/com/microsoft/fluentui/theme/token/controlTokens/AnnouncementCardTokens.kt (hunk 20 above)
Comment (praveen970) [reply fluentui_core/src/main/java/com/microsoft/fluentui/theme/token/controlTokens/AnnouncementCardTokens.kt:24]: using announcementCardInfo will not ovveride the base function.

**Pull request closed without merging**